doc/Language/haskell-to-p6.pod6

```diff
@@ -0,0 +1,420 @@
+=begin pod
+
+=TITLE Perl 6 from Haskell - Nutshell
+
+=SUBTITLE Learning Perl 6 from Haskell, in a nutshell: What do I already know?
+
+Haskell and Perl6 are _very_ different languages. This is obvious. However, that
+does not mean there are not similarities or shared ideas! This page attempts to
+get a Haskell user up and running with Perl6. The Haskell user may find that they need
+not abandon all of their Haskelly thoughts while scripting in Perl6.
+
+Note that this should not be mistaken for a beginner tutorial or overview of Perl
+6; it is intended as a technical reference for Perl 6 learners with a strong
+Haskell background.
+
+
+=head1 Types
+
+=head2 Types vs Values
+
+In Haskell, you have type level programming and then value level programming.
+
+    plusTwo :: Integer -> Integer   -- Types
+    plusTwo x = x + 2               -- Values
+
+You do not mix types and values in Haskell like the below
+
+    plusTwo 2          -- This is valid
+    plusTwo Integer    -- This is not valid
+    
+In Perl6, types (AKA type objects) can live on the same level as values
+
+    sub plus-two(Int $x --> Int) { $x + 2 }
+    
+    plus-two(2)    # This is valid
+    plus-two(Int)  # This is valid
+    
+I will illustrate this unique aspect of Perl6 with one more example:
+
+    multi sub is-string(Str $ --> True) {}
+    multi sub is-string(Any $ --> False) {}
+    
+    is-string('hello')    #True
+    is-string(4)          #False
+
+=head2 Maybe
+
+In Haskell, you have a Maybe type that allows you to forgoe the worry of null types. 
+Let's say you have a hypothetical function that parses a String to an Integer:
+
+    parseInt :: String -> Maybe Integer
+    
+    case parseInt myString of
+      Just x  -> x
+      Nothing -> 0
+
+In Perl6, since type objects coexist with regular objects, we have the concept of Defined
+and Undefined objects. Plain type objects are undefined while instansiated objects are defined.
+
+    sub parse-int(Str $s --> Int) { ... }
+    
+    given parse-int($string) {
+      when * ~~ Int:D { * }
```

Comment:
@CIAvash Good call.

@timo I talked about orelse (which is basically defined-or). But I'm just trying to get this initial draft merged so others can contribute to it.